lib/rubocop/cop/rails/action_order.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop enforces consistent ordering of the standard Rails RESTful
+      # controller actions.
+      #
+      # The cop is configurable and can enforce any ordering of the standard
+      # actions. All other methods are ignored.
+      #
+      # [source,yaml]
+      # ----
+      #  Rails/ActionOrder:
+      #    ExpectedOrder:
+      #      - index
+      #      - show
+      #      - new
+      #      - edit
+      #      - create
+      #      - update
+      #      - destroy
+      # ----
+      #
+      # @example
+      #   # bad
+      #   def index; end
+      #   def destroy; end
+      #   def show; end
+      #
+      #   # good
+      #   def index; end
+      #   def show; end
+      #   def destroy; end
+      class ActionOrder < Base
+        extend AutoCorrector
+
+        MSG = 'Controller actions should be sorted in the following order: %<expected_order>s. ' \
+              'Action `%<current>s` should appear before `%<previous>s`.'
+
+        def_node_search :action_declarations, '(def {%1} ...)'
+
+        def on_class(node)
+          action_declarations(node, actions).each_cons(2) do |previous, current|
+            next if find_index(current) >= find_index(previous)
+
+            register_offense(previous, current)
+          end
+        end
+
+        private
+
+        def expected_order
+          cop_config['ExpectedOrder'].map(&:to_sym)
```

Comment:
Does it make sense to immediately wrap this in a `Set.new`?
Because:
 - why not, one method less
 - protect from misconfiguration when the same action is specified several times

Reply:
That would not work, we need the order intact for the `find_index` method:
```
        def find_index(node)
          expected_order.find_index(node.method_name)
        end
```

Reply:
I'm not sure if it's my access rights or the "allow edits for maintainers" checkbox not ticked, but I couldn't push to this branch.

![image](https://user-images.githubusercontent.com/6916/133319075-d8d464df-7820-470b-9ee5-57eb2117492e.png)

This perfectly fine (apart from `.to_a` since Set doesn't implement `join`).

Leaving it up to you if you like it or not.

Reply:
Ruby gives not guarantees that Sets will remain ordered, so the code above would be unreliable, right? Also, converting to a set, just to convert back to an array seems convoluted to me.

Reply:
You're completely right on both points. Sorry to bother.